### ISSUE TITLE
Secret tracker doesn't disappear after secret triggers

### ISSUE THREAD

**Issue by NightKev** [HearthSim/Hearthstone-Deck-Tracker#2993]
I don't know if the replay will contain any useful info but it's happened a few times, latest was [this game](https://hsreplay.net/replay/MHyjGefhRsGKQnECekaKLB). I noticed that when the secret activated, the secret tracker acted like a new secret was just played.

**Comment by azeier** [comment]
Do you have the hdt log?

**Comment by LufgtHuron** [comment]
Yeah I can confirm this bug too. Secrets dont go away which can become a huge problem after a while.

**Comment by NightKev** [comment]
It's the first game in [this log](https://github.com/HearthSim/Hearthstone-Deck-Tracker/files/628792/hdt_log.-.Copy.txt).

**Comment by azeier** [comment]
Is this problem exclusive to brawl?

**Comment by NightKev** [comment]
I haven't encountered any secrets outside of the Brawl yet, if it happens in a regular game I'll attach another log.

**Comment by azeier** [comment]
I think this might be an issue specific to secrets that were created in the way the brawl does it.

**Comment by riQQ** [comment]
Here an arena log: https://gist.github.com/riQQ/174b7dfe2352ddef47271df7907510dc/

![2016-12-03 11_43_11-hearthstoneoverlay](https://cloud.githubusercontent.com/assets/3404787/20858811/e9181ef0-b94d-11e6-9b15-2f78bafe2864.png)

**Comment by NightKev** [comment]
The brawl probably does it in the same way the Servant of Yogg-Saron card does (which may or may not be the same way Yogg itself does, I think there were some weird bugs with one but not the other?), so if it's brawl specific (though it looks like it's not based on @riQQ's screenshot) then one or the other would probably have issues too.

**Comment by azeier** [comment]
This might be a logging change. Looking into it now.

**Comment by azeier** [comment]
Should be fixed now. 
I think this might be the cause of many of the issues:

```
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -     SHOW_ENTITY - Updating Entity=[name=UNKNOWN ENTITY [cardType=INVALID] id=5 zone=SECRET zonePos=0 cardId= player=1] CardID=EX1_295
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=PREMIUM value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=EXHAUSTED value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=DAMAGE value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=COST value=3
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=CONTROLLER value=1
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=ENTITY_ID value=5
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=SILENCED value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=WINDFURY value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=TAUNT value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=STEALTH value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=DIVINE_SHIELD value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=CHARGE value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=CLASS value=MAGE
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=FACTION value=NEUTRAL
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=CARDTYPE value=SPELL
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=RARITY value=EPIC
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=SECRET value=1
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=FROZEN value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=JUST_PLAYED value=1
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=ZONE_POSITION value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=NUM_ATTACKS_THIS_TURN value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=FORCED_PLAY value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=TO_BE_DESTROYED value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=START_WITH_1_HEALTH value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=CUSTOM_KEYWORD_EFFECT value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=EXTRA_ATTACKS_THIS_TURN value=0
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=TAG_LAST_KNOWN_COST_IN_HAND value=3
D 12:13:26.8145670 PowerTaskList.DebugPrintPower() -         tag=479 value=0```

**Comment by azeier** [comment]
Here is a build with the fix: https://github.com/azeier/HDT-Test/releases/download/v1.1.2.2/Hearthstone.Deck.Tracker-v1.1.2.2.zip

Can you guys confirm its fixed?

**Comment by codename47-vin** [comment]
Just tested in brawl. Worked well for me. Thank you!

**Comment by azeier** [comment]
Fixed with v1.1.3
